src/components/base/JIcon/JIcon.js

```diff
@@ -1,33 +1,53 @@
 // @flow
 
-import React, { PureComponent } from 'react'
+import React from 'react'
+import classNames from 'classnames'
+import { keyBy } from 'lodash-es'
 
-export type JIconSize = 'small' | 'medium' | 'large' | 'xlarge'
 export type JIconColor = 'white' | 'blue' | 'gray' | 'sky' | 'red' | 'black'
 
 type Props = {
   name: string,
-  size: JIconSize,
-  color: JIconColor,
+  color: ?JIconColor,
 }
 
-class JIcon extends PureComponent<Props> {
-  static defaultProps = {
-    size: 'medium',
-    color: 'white',
-  }
-
-  render() {
-    const {
-      name,
-      size,
-      color,
-    }: Props = this.props
-
-    return (
-      <div className={`j-icon -${name} -${size} -${color}`} />
-    )
-  }
+const files = require.context('../../../public/assets/icons/sprite-pack', true, /.*\.svg$/)
```

Comment:
Maybe, we can to move this "dirty" part of code to other file and call them like Hight-Order Function:
```javascript
import JIcon from '...'
function importIconsFiles (props: Props) {
  /*
    dirty part with imports files
  */

  return <JIcon {...props} />
}
```